### PR TITLE
DSCP based marking and queue selection for tunnels and PFC frames

### PR DESCRIPTION
**Pull request by JaiOCP**
Following attributes are introduced for fine grain marking and queue selection for tunneled traffic and PFC frames.
Same attributes are also added to saiswitch.

Tunnel Encap:
SAI_TUNNEL_ATTR_ENCAP_QOS_DSCP_TO_TC_MAP: For DSCP based TC MAP for queue
SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP: for Remarking
SAI_TUNNEL_ATTR_ENCAP_QOS_PFC_TC_TO_PRIORITY_GROUP_MAP: Priority group selection for PFC frames

Tunnel Decap:
SAI_TUNNEL_ATTR_DECAP_QOS_DSCP_TO_TC_MAP: For DSCP based TC MAP for queue
SAI_TUNNEL_ATTR_DECAP_QOS_DSCP_TO_PFC_TC_MAP: For DSCP based TC MAP for PFC frames
SAI_TUNNEL_ATTR_DECAP_QOS_PFC_TC_TO_PRIORITY_GROUP_MAP: Priority group selection for PFC frames

### CODE REVIEW

File: inc/saiswitch.h [hunk 1]
```diff
@@ -514,6 +514,78 @@ typedef enum _sai_switch_tunnel_attr_t
...
+     * Map id = #SAI_NULL_OBJECT_ID to disable map on port.
+     * Default no map
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_QOS_MAP
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_SWITCH_TUNNEL_ATTR_DECAP_QOS_PFC_TC_TO_PRIORITY_GROUP_MAP,
```
Comment (lguohan) [inline inc/saiswitch.h:587]: SAI_SWITCH_TUNNEL_ATTR_DECAP_QOS_PFC_TC_TO_PRIORITY_GROUP_MAP -> SAI_SWITCH_TUNNEL_ATTR_DECAP_QOS_TC_TO_PRIORITY_GROUP_MAP

File: inc/saiswitch.h [hunk 2]
```diff
@@ -514,6 +514,78 @@ typedef enum _sai_switch_tunnel_attr_t
...
+    /**
+     * @brief Enable DSCP -> PFC TC MAP on tunnel at termination (Network-to-access) node. Used to flexibly map DSCP to TC for PFC
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_QOS_MAP
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_SWITCH_TUNNEL_ATTR_DECAP_QOS_DSCP_TO_PFC_TC_MAP,
```
Comment (lguohan) [inline inc/saiswitch.h:573]: remove this

File: inc/saiswitch.h (hunk 1 above)
Comment (lguohan) [reply inc/saiswitch.h:587]: add SAI_SWITCH_TUNNEL_ATTR_DECAP_QOS_TC_TO_QUEUE_MAP

File: inc/saiswitch.h [hunk 3]
```diff
@@ -514,6 +514,78 @@ typedef enum _sai_switch_tunnel_attr_t
...
+     * Map id = #SAI_NULL_OBJECT_ID to disable map on port.
+     * Default no map
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_QOS_MAP
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_SWITCH_TUNNEL_ATTR_ENCAP_QOS_PFC_TC_TO_PRIORITY_GROUP_MAP,
```
Comment (lguohan) [inline inc/saiswitch.h:551]: SAI_SWITCH_TUNNEL_ATTR_ENCAP_QOS_TC_TO_PRIORITY_GROUP_MAP

**Comment by JaiOCP** [comment]
Enum Values are shifted so opening a new PR instead of quashing

**Pull request closed without merging**